src/Features/Core/Portable/MakeMemberStatic/AbstractMakeMemberStaticCodeFixProvider.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.MakeMemberStatic
+{
+    internal abstract class AbstractMakeMemberStaticCodeFixProvider
+        : SyntaxEditorBasedCodeFixProvider
+    {
+        internal sealed override CodeFixCategory CodeFixCategory => CodeFixCategory.Compile;
+
+        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            if (context.Diagnostics.Length != 1 ||
```

Comment:
You don't need diagnostics count check and don't need FindNode check over here. Please remove this if statement.

Reply:
If I recall correctly the tests I did, this check was still useful for fix-all cases because the other register wasn't called back.

Reply:
I confirm that there are some issue with the fix-all when removing the checks.